Meadowlark-js/packages/meadowlark-core/src/validation/DocumentIdentityExtractor.ts

```diff
@@ -95,11 +97,20 @@ function documentIdentitiesFrom(
   return result;
 }
 
+/**
+ * All descriptor documents have the same identity fields
+ */
+function descriptorDocumentIdentityWithSecurity(body: DescriptorDocument): DocumentIdentityWithSecurity {
+  return { documentIdentity: descriptorDocumentIdentityFrom(body), studentId: null, edOrgId: null };
+}
+
 /**
  * Takes a MetaEd entity object and a API JSON body for the resource mapped to that MetaEd entity and
  * extracts the document identity information from the JSON body. Also extracts security information, if any.
  */
 export function extractDocumentIdentity(entity: TopLevelEntity, body: object): DocumentIdentityWithSecurity {
+  if (entity.type === 'descriptor') return descriptorDocumentIdentityWithSecurity(body as DescriptorDocument);
+
```

Comment:
If it's a descriptor, the body's been schema validated at this point, we know it's a DescriptorDocument